File: 4080-smallest-missing-multiple-of-k/smallest-missing-multiple-of-k.java
class Solution {
    public int missingMultiple(int[] nums, int k) {
        int n = nums.length;
        HashSet<Integer> un = new HashSet<>();
        for(int i =0 ;i < n;i++){
            un.add(nums[i]);
        }
        n = k;
        while(un.contains(n)){
            n+=k;
        }
        return n;
    }
}

File: 4080-smallest-missing-multiple-of-k/smallest-missing-multiple-of-k.cpp

class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        int n = nums.size();
        set<int> un;
        for(int i = 0;i < n;i++){
            un.insert(nums[i]);
        }
        n = 1;
        while(true){
            if(!un.contains(n*k)) return n*k;
            n++;
        }
        return 0;
    }
};